src/Sulu/Bundle/AdminBundle/Resources/js/components/Number/tests/Number.test.js

```diff
@@ -1,40 +1,45 @@
 // @flow
 import React from 'react';
-import {render, shallow} from 'enzyme';
+import {fireEvent, render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import Number from '../Number';
 
 test('Number should render', () => {
-    expect(render(<Number onChange={jest.fn()} value={undefined} />)).toMatchSnapshot();
+    const {container} = render(<Number onChange={jest.fn()} value={undefined} />);
+    expect(container).toMatchSnapshot();
 });
 
 test('Number should render when disabled', () => {
-    expect(render(<Number disabled={true} onChange={jest.fn()} value={8} />)).toMatchSnapshot();
+    const {container} = render(<Number disabled={true} onChange={jest.fn()} value={8} />);
+    expect(container).toMatchSnapshot();
 });
 
-test('Number should call onChange with parsed value', () => {
+test('Number should call onChange with parsed value', async() => {
     const onChange = jest.fn();
-    const number = shallow(<Number onChange={onChange} value={undefined} />);
+    render(<Number onChange={onChange} value={2} />);
 
-    const event = {};
-    number.find('Input').simulate('change', '10.2', event);
+    const input = screen.queryByDisplayValue(2);
+    fireEvent.keyDown(input, {target: {value: '10.2'}});
```

Comment:
Think we have a problem here to go with `10.` to `10.2` /cc @niklasnatter any idea?

Reply:
https://github.com/testing-library/user-event/issues/549 recommends to add a wrapper component for these components. i will have a look at that 🙂  
another workaround would be using `paste` instead of `type` to change multiple characters in a single event. but i think that will confuse developers because it is not intuitive to understand why we used `paste`

Reply:
i merged this PR using `paste` and added a helper function to wrap the component in https://github.com/sulu/sulu/pull/6795